From 145c61ffd41d1f4ad821a15d0268fee2c1310011 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Mon, 7 May 2018 14:20:00 +0200
Subject: Wrap mksnapshot and torque calls on Yocto building with qemu.

The patch below makes the V8 binaries run during the build be invoked through
QEMU, as they are built for the target.

Based on original patch from Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>

Upstream-Status: Inappropriate [embedded specific]

diff --git a/BUILD.gn b/BUILD.gn
index 9992bf6098..165855a82c 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -879,6 +879,7 @@ action("run_torque") {
   ]
 
   args = [
+    "./v8-qemu-wrapper.sh",
     "./" + rebase_path(get_label_info(":torque($v8_torque_toolchain)",
                                       "root_out_dir") + "/torque",
                        root_build_dir),
@@ -923,6 +924,7 @@ template("run_mksnapshot") {
     data = []
 
     args = [
+      "./v8-qemu-wrapper.sh",
       "./" + rebase_path(get_label_info(":mksnapshot($v8_snapshot_toolchain)",
                                         "root_out_dir") + "/mksnapshot",
                          root_build_dir),
-- 
2.14.1

